doc/extensions/best-practices.rst

```diff
@@ -78,6 +78,33 @@ available in CKAN for this purpose:
 
     ckan db downgrade -p PLUGIN_NAME
 
+------------------------------------
+Declare models using shared metadata
+------------------------------------
+
+Use :py:obj:`ckan.model.meta.metadata` object for any new SQLAlchemy model
+registered via extension. It helps SQLAlchemy to resolve cascade relationships
+and control orphan removals. In addition, ``clean_db`` test fixture cleans
+tables for every model registered using :py:obj:`ckan.model.meta.metadata`, so
+new models are integrated into the test suite without additional efforts.
+
+Example::
+
+    import ckan.model as model
+    from sqlalchemy.ext.declarative import declarative_base
+
+    Base = declarative_base(metadata=model.meta.metadata)
+
+    class ExtModel(Base):
+        __tablename__ = "ext_model"
+        id = Column(String(50), primary_key=True)
+        ...
+
+
+.. note:: Starting from CKAN v2.10.0 the :py:class:`~ckan.model.base.Base`
+          class for declarative-style models, that is already attached to the
+          metadata object, can be imported from :py:mod:`ckan.model.base`.
```

Comment:
@smotornyuk to further simplify things, what do you think about exposing `ckan.model.base.Base` as `toolkit.BaseModel`? This way users don't need to import stuff from `ckan.model` and we can document it in the [toolkit's reference](https://docs.ckan.org/en/latest/extensions/plugins-toolkit.html) it's probably easier to find for extension developers:

```python
from ckan.plugins import toolkit

class ExtModel(toolkit.BaseModel):
    __tablename__ = "ext_model"
    id = Column(String(50), primary_key=True)
    ...
```

Reply:
I like it. It felt kind of bad to ask users using something from such depths of CKAN and rely on internal module structure, but as usual, I completely forgot about the toolkit:)